lib/hydra/works/services/generic_file/full_text_extraction_service.rb

```diff
@@ -0,0 +1,33 @@
+module Hydra::Works
+  # Extract the full text from the content using Solr's extract handler
+  class FullTextExtractionService
+    def self.run(generic_file)
```

Comment:
This is a good start. Could you reformat it as YARD? See here for an example: https://github.com/projecthydra/hydra-collections/pull/84/files#diff-68cd07172cdadf2489d69b510b06b313R23